Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -303,59 +318,83 @@ public object Deserialize(IIonReader reader, Type type, IonType ionType)
+                    return serializer.Deserialize(reader);
+                }
+                else
+                {
+                    var serializer = this.GetCustomPrimitiveSerializer<double>() ?? new IonDoubleSerializer();
+                    return serializer.Deserialize(reader);
                 }
-                return new IonDoubleSerializer().Deserialize(reader);
             }
 
             if (ionType == IonType.Decimal)
             {
                 if (reader.GetTypeAnnotations().Any(s => s.Equals(IonDecimalSerializer.ANNOTATION)))
                 {
-                    return new IonDecimalSerializer().Deserialize(reader);
+                    var serializer = this.GetCustomPrimitiveSerializer<decimal>() ?? new IonDecimalSerializer();
+                    return serializer.Deserialize(reader);
+                }
+                else
+                {
+                    var serializer = this.GetCustomPrimitiveSerializer<BigDecimal>() ?? new IonBigDecimalSerializer();
+                    return serializer.Deserialize(reader);
                 }
-                return new IonBigDecimalSerializer().Deserialize(reader);
             }
 
             if (ionType == IonType.Blob) 
             {
                 if (reader.GetTypeAnnotations().Any(s => s.Equals(IonGuidSerializer.ANNOTATION))
                     || type.IsAssignableTo(typeof(Guid)))
                 {
-                    return new IonGuidSerializer(options).Deserialize(reader);
+                    var serializer = this.GetCustomPrimitiveSerializer<Guid>() ?? new IonGuidSerializer(options);
+                    return serializer.Deserialize(reader);
+                }
+                else
+                {
+                    var serializer = this.GetCustomPrimitiveSerializer<byte[]>() ?? new IonByteArraySerializer();
+                    return serializer.Deserialize(reader);
                 }
-                return new IonByteArraySerializer().Deserialize(reader);
             }
 
             if (ionType == IonType.String) 
             {
-                return new IonStringSerializer().Deserialize(reader);
+                var serializer = this.GetCustomPrimitiveSerializer<string>() ?? new IonStringSerializer();
+                return serializer.Deserialize(reader);
             }
 
             if (ionType == IonType.Symbol) 
             {
-                return new IonSymbolSerializer().Deserialize(reader);
+                var serializer = this.GetCustomPrimitiveSerializer<SymbolToken>() ?? new IonSymbolSerializer();
+                return serializer.Deserialize(reader);
             }
 
             if (ionType == IonType.Timestamp) 
             {
-                return new IonDateTimeSerializer().Deserialize(reader);
+                var serializer = this.GetCustomPrimitiveSerializer<DateTime>() ?? new IonDateTimeSerializer();

```

Comment:
I would probably pass in the default serializer to the `GetCustomPrimitiveSerializer` method.

Reply:
Addressed.